extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -197,88 +197,97 @@ export class ResourceTypePickerDialog extends DialogBase {
-					}
-
-					return [tool.displayName, tool.description, tool.displayStatus, tool.fullVersion || '', toolRequirement.version || '', tool.installationPath || ''];
-				});
+			let toolsLoadingErrors: string[] = [];
+			Promise.all(this._tools.map(tool => tool.loadInformation().catch(() => {
+				toolsLoadingErrors.push(`${tool.displayName}:${tool.statusDescription!}`);
+			})))
+				.then(() => this.executeToolsTableWorkflow(currentRefreshTimestamp, toolsLoadingErrors))
+				.catch(() => {/* Empty catch to keep linter happy */ });
+		}
+	}
 
-				this._installToolButton.hidden = minVersionCheckFailed || (toolsToAutoInstall.length === 0);
-				this._dialogObject.okButton.enabled = messages.length === 0 && !minVersionCheckFailed && (toolsToAutoInstall.length === 0);
-				if (messages.length !== 0) {
-					if (messages.length > 1) {
-						messages = messages.map(message => `•	${message}`);
-					}
-					messages.push(localize('deploymentDialog.VersionInformationDebugHint', "You will need to restart Azure Data Studio if the tools are installed manually after Azure Data Studio is launched to pick up the updated PATH environment variable. You may find additional details in 'Deployments' output channel"));
-					this._dialogObject.message = {
-						level: azdata.window.MessageLevel.Error,
-						text: messages.join(EOL)
-					};
-				} else if ((toolsToAutoInstall.length !== 0) && !this._installationInProgress) {
-					const installationNeededHeader = toolsToAutoInstall.length === 1
-						? localize('deploymentDialog.InstallToolsHintOne', "Tool: {0} is not installed, you can click the \"{1}\" button to install it.", toolsToAutoInstall[0].displayName, this._installToolButton.label)
-						: localize('deploymentDialog.InstallToolsHintMany', "Tools: {0} are not installed, you can click the \"{1}\" button to install them.", toolsToAutoInstall.map(t => t.displayName).join(', '), this._installToolButton.label);
-					let infoText: string[] = [installationNeededHeader];
-					const informationalMessagesArray = this._tools.reduce<string[]>((returnArray, currentTool) => {
-						if (currentTool.autoInstallNeeded) {
-							returnArray.push(...currentTool.dependencyMessages);
-						}
-						return returnArray;
-					}, /* initial Value of return array*/[]);
-					const informationalMessagesSet: Set<string> = new Set<string>(informationalMessagesArray);
-					if (informationalMessagesSet.size > 0) {
-						infoText.push(...informationalMessagesSet.values());
-					}
-					// we don't have scenarios that have mixed type of tools - either we don't support auto install: docker, or we support auto install for all required tools
-					this._dialogObject.message = {
-						level: azdata.window.MessageLevel.Warning,
-						text: infoText.join(EOL)
-					};
+	private async executeToolsTableWorkflow(currentRefreshTimestamp: number, toolsLoadingErrors: string[]): Promise<void> {
```

Comment:
It doesn't look like there's any need for this to be async - am I missing something?